src/bio/SConscript

```diff
@@ -14,22 +14,17 @@ def scons():
     SConscript('smd/SConscript')
     denv.AppendUnique(LIBPATH=['smd'])
 
-    # Link to DPDK static libs
-    denv.AppendUnique(LINKFLAGS=['-Wl,--whole-archive', \
-		      '-lrte_mempool', '-lrte_mempool_ring', '-lrte_bus_pci', \
-		      '-lrte_pci', '-lrte_ring', '-lrte_mbuf', '-lrte_eal', \
-		      '-lrte_kvargs', \
-		      '-Wl,--no-whole-archive'])
-
-    # SPDK shared libs
+    # SPDK related libs
     libs = ['spdk_env_dpdk', 'spdk_thread', 'spdk_bdev', 'spdk_copy']
+    libs += ['rte_mempool', 'rte_mempool_ring', 'rte_bus_pci']
+    libs += ['rte_pci', 'rte_ring', 'rte_mbuf', 'rte_eal', 'rte_kvargs']
     libs += ['spdk_bdev_aio', 'spdk_bdev_nvme', 'spdk_bdev_malloc']
     libs += ['spdk_conf', 'spdk_blob', 'spdk_nvme', 'spdk_util']
     libs += ['spdk_json', 'spdk_jsonrpc', 'spdk_rpc', 'spdk_trace']
     libs += ['spdk_sock', 'spdk_log', 'spdk_notify', 'spdk_blob_bdev']
 
     # Other libs
-    libs += ['numa', 'smd']
+    libs += ['spdk', 'numa', 'dl', 'smd']
```

Comment:
We've linked dependent individual spdk, dpdk libs, 'spdk' isn't necessary anymore, it's just a script contains all the spdk libs.